File: solution/python3/83.remove-duplicates-from-sorted-list.py
#
# @lc app=leetcode id=83 lang=python3
#
# [83] Remove Duplicates from Sorted List
#

# @lc code=start
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        dummy = ListNode(-101)
        current = dummy
        while True:
            if not head:
                break
            if current.val < head.val:
                current.next = ListNode(head.val)
                current = current.next
            head = head.next
        return dummy.next




# @lc code=end

File: solution/java8/83.remove-duplicates-from-sorted-list.java
/*
 * @lc app=leetcode id=83 lang=java
 *
 * [83] Remove Duplicates from Sorted List
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null) {
            return null;
        }

        ListNode dummy = new ListNode(-101);
        ListNode current = dummy;
        while (true) {
            System.out.println(head.val);
            if (head.next == null) {
                break;
            }
            if (head.val != current.val) {
                current.next = new ListNode(head.val);
                current = current.next;
            }
            head = head.next;
        }
        if (head.val != current.val) {
            current.next = new ListNode(head.val);
            current = current.next;
        }

        return dummy.next;

    }
}
// @lc code=end

